### PR TITLE
Add SizeChecker for SV

### PR DESCRIPTION
**Pull request by NympheaR**

### CODE REVIEW

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*  Mass Release
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.h:1]: change the comment here

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+
+void SizeChecker::enter_check_mode(ConsoleHandle& console, BotBaseContext& context){
+    WallClock start = current_time();
+
+    while (true){
+        if (current_time() - start > std::chrono::minutes(5)){
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, console,
+                "enter_check_mode(): Failed to enter box mode after 5 minutes.",
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp:90]: Why does the timeout need to be that huge ?

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+}
+
+
+
+void SizeChecker::exit_check_mode(ConsoleHandle& console, BotBaseContext& context){
+    WallClock start = current_time();
+
+    while (true){
+        if (current_time() - start > std::chrono::minutes(5)){
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp:142]: same, why does it needs to be that huge

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+                pbf_press_button(context, BUTTON_A, 20, 20);
+
+                // If last cell not empty.
+                if (row == 4 && col == 5){
+                    exit_check_mode(env.console, context);
+                } else {
+                    enter_check_mode(env.console, context);
+                    // Default state on box entry does not highlight selection, move cursor so it does.
+                    pbf_press_dpad(context, DPAD_LEFT, 5, 100);
+                }
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp:243]: I don't understand this part, do you mind explaining ?

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.h (hunk 1 above)
Comment (NympheaR) [reply SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.h]: Changed.

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp (hunk 2 above)
Comment (NympheaR) [reply SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp]: Reduced to 2 min.

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp (hunk 3 above)
Comment (NympheaR) [reply SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp]: Reduced to 2 min.

File: SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp (hunk 4 above)
Comment (NympheaR) [reply SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp]: I agree, this is too hacky. I've updated the overall logic to be more intuitive now so it should make more sense.

**Review by pifopi — APPROVED** [review]

**Pull request merged**